apisix/balancer.lua

```diff
@@ -87,12 +87,17 @@ local function create_checker(upstream, healthcheck_parent)
     if healthcheck == nil then
         healthcheck = require("resty.healthcheck")
     end
-    local checker = healthcheck.new({
+    local checker, err = healthcheck.new({
```

Comment:
Is it possible to write a test for it?

Reply:
I remember that error happened when more workers tried to create checkers at the same moment. In the PROD, I have encountered such problems. I will consider how to add test cases， but I think it's a bit difficult :(

Reply:
OK

Reply:
I can't find a way to reproduce it by using Test::Nginx.  In my test case, I've set master_on,workers(10) and added a server on different port, I create many threads to connect that server when got the initial request, and wanted the different workers would handled that connections, but  only the same worker  which got  the initial request dealt with the connection requests.

@membphis  @spacewander  Do you have a way to get test requests on different workers?

Reply:
AFAIK, there is no such a way.